### PR TITLE
Feature/improve browsing source performance

### PR DESCRIPTION
**Pull request by schroda**

### CODE REVIEW

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt [hunk 1]
```diff
@@ -50,35 +45,38 @@ object MangaList {
...
-                        (MangaTable.url eq manga.url) and (MangaTable.sourceReference eq sourceId)
-                    }.firstOrNull()
-                if (mangaEntry == null) { // create manga entry
-                    val mangaId =
-                        MangaTable.insertAndGetId {
-                            it[url] = manga.url
-                            it[title] = manga.title
+            val existingMangaUrlsToId =
+                MangaTable.slice(MangaTable.url, MangaTable.id).select {
+                    MangaTable.url inList mangas.map { it.url }
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt]: Source id needs to be taken into account.
```suggestion
                    MangaTable.sourceReference eq sourceId and (MangaTable.url inList mangas.map { it.url })
```

**Review by Syer10 — CHANGES_REQUESTED** [review]

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt (hunk 1 above)
Comment (schroda) [reply server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt]: fixed

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt [hunk 2]
```diff
@@ -50,35 +46,38 @@ object MangaList {
...
-                    }.firstOrNull()
-                if (mangaEntry == null) { // create manga entry
-                    val mangaId =
-                        MangaTable.insertAndGetId {
-                            it[url] = manga.url
-                            it[title] = manga.title
+            val existingMangaUrlsToId =
+                MangaTable.slice(MangaTable.url, MangaTable.id).select {
+                    (MangaTable.sourceReference eq sourceId) and (MangaTable.url inList mangas.map { it.url })
+                }.map { Pair(it[MangaTable.url], it[MangaTable.id].value) }
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt]:
```suggestion
                }.associate { Pair(it[MangaTable.url], it[MangaTable.id].value) }
```

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt [hunk 3]
```diff
@@ -50,35 +46,38 @@ object MangaList {
...
+                    this[MangaTable.status] = it.status
+                    this[MangaTable.thumbnail_url] = it.thumbnail_url
+                    this[MangaTable.updateStrategy] = it.update_strategy.name
 
-                    mangaId
-                } else {
-                    mangaEntry[MangaTable.id].value
-                }
+                    this[MangaTable.sourceReference] = sourceId
+                }.map { Pair(it[MangaTable.url], it[MangaTable.id].value) }
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt]:
```suggestion
                }.associate { Pair(it[MangaTable.url], it[MangaTable.id].value) }
```

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt [hunk 4]
```diff
@@ -50,35 +46,38 @@ object MangaList {
...
+
+            // delete thumbnail in case cached data still exists
+            insertedMangaUrlsToId.forEach { Manga.clearThumbnail(it.second) }
+
+            val mangaUrlsToId = existingMangaUrlsToId + insertedMangaUrlsToId
+
+            mangas.map { manga ->
+                mangaUrlsToId.find { (url) -> url == manga.url }?.second
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt]:
```suggestion
                mangaUrlsToId[manga.url]
```

File: server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt [hunk 5]
```diff
@@ -50,35 +46,38 @@ object MangaList {
...
 
-                    mangaId
-                } else {
-                    mangaEntry[MangaTable.id].value
-                }
+                    this[MangaTable.sourceReference] = sourceId
+                }.map { Pair(it[MangaTable.url], it[MangaTable.id].value) }
+
+            // delete thumbnail in case cached data still exists
+            insertedMangaUrlsToId.forEach { Manga.clearThumbnail(it.second) }
```
Comment (Syer10) [inline server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt]:
```suggestion
            insertedMangaUrlsToId.forEach { _, id -> Manga.clearThumbnail(id) }
```

**Review by Syer10 — CHANGES_REQUESTED** [review]

**Review by Syer10 — APPROVED** [review]

**Pull request merged**